### PR TITLE
Add caps lock warning to the password textbox

### PR DESCRIPTION
**Pull request by jorolf**

### CODE REVIEW

File: osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs [hunk 1]
```diff
@@ -51,5 +62,36 @@ protected override void LoadComplete()
...
+                Colour = colour.YellowLight;
+            }
+
+            protected override void Update()
+            {
+                base.Update();
+                updateVisibility();
+            }
+
+            private void updateVisibility() => FadeTo(Console.CapsLock ? 1 : 0, 250, EasingTypes.OutQuint);
```
Comment (peppy) [inline osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs]: you can't do this with transform helpers. you need to store a state and run them only once, else they are going to run far longer than you expect.

**Comment by peppy** [comment]
https://github.com/mono/mono/blob/master/mcs/class/corlib/System/Console.cs#L556

Checking for caps lock needs to be done as a per-platform implementation. See other examples of how this is done in the framework.

**Review by peppy — CHANGES_REQUESTED** [review]
as above

**Comment by Jorolf** [comment]
closed until I find a better way to test if capslock is enabled

**Pull request closed without merging**